asciidoc-confluence-publisher-converter/src/main/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluenceConverter.java

```diff
@@ -56,58 +53,124 @@
-            buildPageTree(confluencePublisherMetadata.getPages(), confluencePublisherMetadataRegistry, generatedDocOutputPath);
+            return confluencePublisherMetadata;
+        } catch (Exception e) {
+            throw new RuntimeException("Could not convert asciidoc pages", e);
         }
+    }
+
+    private static List<ConfluencePageMetadata> buildPageTree(Path templatesRootFolder, Path assetsRootFolder, List<AsciidocPage> asciidocPages) {
+        List<ConfluencePageMetadata> confluencePages = new ArrayList<>();
+
+        asciidocPages.forEach((asciidocPage) -> {
+            try {
+                Path pageAssetsFolder = determinePageAssetsFolder(assetsRootFolder, asciidocPage);
+                createDirectories(pageAssetsFolder);
+
+                AsciidocConfluencePage asciidocConfluencePage = newAsciidocConfluencePage(asciidocPage, templatesRootFolder, pageAssetsFolder);
+                Path contentFileTargetPath = writeToTargetStructure(asciidocPage, pageAssetsFolder, asciidocConfluencePage);
+
+                List<AttachmentMetadata> attachments = buildAttachments(asciidocPage, pageAssetsFolder, asciidocConfluencePage.attachments());
+                copyAttachmentsAvailableInSourceStructureToTargetStructure(attachments);
+
+                List<ConfluencePageMetadata> childConfluencePages = buildPageTree(templatesRootFolder, assetsRootFolder, asciidocPage.children());
+
+                ConfluencePageMetadata confluencePageMetadata = new ConfluencePageMetadata();
+                confluencePageMetadata.setTitle(asciidocConfluencePage.pageTitle());
+                confluencePageMetadata.setContentFilePath(contentFileTargetPath.toAbsolutePath().toString());
+                confluencePageMetadata.setChildren(childConfluencePages);
+                confluencePageMetadata.getAttachments().putAll(toTargetAttachmentFileNameAndAttachmentPath(attachments));
 
-        return confluencePublisherMetadata;
+                confluencePages.add(confluencePageMetadata);
+            } catch (IOException e) {
+                throw new RuntimeException("Could not convert and build confluence page", e);
+            }
+        });
+
+        return confluencePages;
+    }
+
+    private static List<AttachmentMetadata> buildAttachments(AsciidocPage asciidocPage, Path pageAssetsFolder, Map<String, String> attachmentsWithRelativePath) {
+        return attachmentsWithRelativePath.keySet().stream()
+                .map((attachmentWithRelativePath) -> {
+                    Path relativeAttachmentPath = Paths.get(attachmentWithRelativePath);
+                    Path attachmentSourcePath = asciidocPage.path().getParent().resolve(relativeAttachmentPath);
+                    Path attachmentTargetPath = pageAssetsFolder.resolve(relativeAttachmentPath.getFileName());
+
+                    return new AttachmentMetadata(attachmentSourcePath, attachmentTargetPath);
+                }).collect(toList());
```

Comment:
`.collect(toList());` should be on  new line.

Reply:
Fixed in d7f0bd2b5a6b41b75003e8d25a18e4bbb905a721